src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -448,63 +425,60 @@ class npc_void_sentinel : public CreatureScript
 public:
     npc_void_sentinel() : CreatureScript("npc_void_sentinel") { }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return new npc_void_sentinelAI(creature);
-    }
-
     struct npc_void_sentinelAI : public ScriptedAI
     {
         npc_void_sentinelAI(Creature* creature) : ScriptedAI(creature)
         {
-            Initialize();
+            _instance = me->GetInstanceScript();
+            me->SetReactState(REACT_PASSIVE);
         }
 
-        void Initialize()
+        void Reset() override
         {
-            PulseTimer = 3000;
-            VoidBlastTimer = 45000; //is this a correct timer?
-        }
+            if (Creature* muru = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_MURU)))
+                muru->AI()->JustSummoned(me);
```

Comment:
Alright, this is fine then.

Some of these muru lookups in ObjectAccessor could use some TC_LOG_WARN when not found.